packages/office-ui-fabric-react/src/components/Modal/Modal.base.tsx

```diff
@@ -140,21 +140,14 @@ export class ModalBase extends BaseComponent<IModalProps, IDialogState> implemen
       hasBeenOpened,
       modalRectangleTop,
       topOffsetFixed,
-      isModeless
+      isModeless,
+      layerProps
     });
 
-    // if the modal is modeless, add the classname to correctly style the layer
-    const layerClassName = isModeless
-      ? this.props.className
-        ? `${this.props.className} ${classNames.layer}`
-        : classNames.layer
-      : this.props.className;
-
     const mergedLayerProps = {
       ...DefaultLayerProps,
       ...this.props.layerProps,
       onLayerDidMount: layerProps && layerProps.onLayerDidMount ? layerProps.onLayerDidMount : onLayerDidMount,
-      className: layerClassName,
```

Comment:
> className: layerClassName, [](start = 5, length = 27)

doesn't classNames.layer still need to be be passed down to the layer?

You could make the getClassNames always set layerProps && layerProps.className on layer and then add the additional styling if it isModeless. If you did that you could just always say className: classNames.layer here.

Reply:
@jspurlin thanks for the comment. That is the implementation I had in this [commit](https://github.com/OfficeDev/office-ui-fabric-react/pull/8273/commits/7e0229812231db1d891af20f8f8b1c6df33bf87a). But got a suggestion about doing the styling in the styles file, `Modal.styles.ts`. Please advise on whether the changes there look okay.

Reply:
So you are setting the styling in Modal.styles.ts (which updates the className so that it includes the original classname plus potentially the new classname) but you still need to set className to classNames.layer so that it can use that updated value